tests/Oasis/SignerTests.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Oasis/Signer.h"
+#include "Oasis/Address.h"
+#include "HexCoding.h"
+#include "PrivateKey.h"
+#include "PublicKey.h"
+
+#include <gtest/gtest.h>
+
+using namespace TW;
+using namespace TW::Oasis;
+
+TEST(OasisSigner, Sign) {
+    auto input = Proto::SigningInput();
+    auto &transfer = *input.mutable_transfer();
+
+    Proto::Message::MethodType methodType = Proto::Message_MethodType_staking;
+    Proto::Message::OperationType operationType = Proto::Message_OperationType_Transfer;
+    std::string requestMethodString = Message_MethodType_Name(methodType) + "." + Message_OperationType_Name(operationType); // "stacking.Transfer"
```

Comment:
```suggestion
    // make this const
    std::string requestMethodString = "stacking.Transfer";
```

Sorry for overlooking, this one is superfluous, besides that the same hardcode string is also used in Swift and Kotlin code.

Let's do this:
1. Remove `method` from `TransferMessage` in `Oasis.proto`, we know which string to use here, if we want to add other message like staking, we can add another `StakingMessage` and another `staking.XXXX` string
2. Remove `OperationType` and `MethodType` from `Oasis.proto`

Reply:
It's ok. I removed that lines.